File: sudoku-common/common-core/src/main/java/com/zumin/sudoku/common/core/constant/PermissionConstants.java
package com.zumin.sudoku.common.core.constant;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * 权限常量类
 */
public interface PermissionConstants {

  /**
   * 管理员名
   */
  String ADMIN_NAME = "ROLE_ADMIN";

  /**
   * 用户名
   */
  String USER_NAME = "ROLE_USER";

  /**
   * 用户角色名
   */
  List<String> USER_ROLE_NAME = Collections.singletonList(USER_NAME);

  /**
   * 管理员角色名
   */
  List<String> ADMIN_ROLE_NAME = Arrays.asList(ADMIN_NAME, USER_NAME);

}

File: sudoku-common/common-core/src/main/java/com/zumin/sudoku/common/core/auth/PermissionConstants.kt
package com.zumin.sudoku.common.core.auth

/**
 * 管理员名
 */
const val ADMIN_NAME = "ROLE_ADMIN"

/**
 * 用户名
 */
const val USER_NAME = "ROLE_USER"

/**
 * 用户角色名
 */
val USER_ROLE_NAME = listOf(USER_NAME)

/**
 * 管理员角色名
 */
val ADMIN_ROLE_NAME = listOf(ADMIN_NAME, USER_NAME)
